docs/sources/reference/components/otelcol.receiver.file_stats.md

```diff
@@ -0,0 +1,269 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol.receiver.file_stats/
+title: otelcol.receiver.file_stats
+description: Learn about otelcol.receiver.file_stats
+---
+
+# otelcol.receiver.file_stats
+
+`otelcol.receiver.file_stats` collects metrics from files specified with a glob pattern.
+
+{{< admonition type="note" >}}
+`otelcol.receiver.file_stats` is a wrapper over the upstream OpenTelemetry Collector `filestats` receiver from the `otelcol-contrib` distribution.
+Bug reports or feature requests will be redirected to the upstream repository, if necessary.
+{{< /admonition >}}
+
+Multiple `otelcol.receiver.file_stats` components can be specified by giving them different labels.
+
+## Usage
+
+```alloy
+otelcol.receiver.file_stats "LABEL" {
+  include = "GLOB_PATTERN"
+
+  output {
+    metrics = [...]
+  }
+}
+```
+
+## Arguments
+
+`otelcol.receiver.file_stats` supports the following arguments:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`include` | `string` | Glob path for files to collect stats from. | | yes
+`collection_interval` | `duration` | How often to collect file information. | `"1m"` | no
+`initial_delay` | `duration` | Initial time to wait before collecting file information. | `"1s"` | no
+`timeout` | `duration` | Timeout for collecting file information; `0s` means no timeout. | `"0s"` | no
```

Comment:
Is this per file or for a whole update task?

Reply:
Good point, I'll check and update the docs

Reply:
It's the timeout for the entire poll of all files matched by `include`